src/plugins/signoff/components.js

```diff
@@ -0,0 +1,267 @@
+  // We use the source last status change as review request datetime.
+  let link = "disabled";
+  let {lastStatusChanged: lastRequested} = source;
+  if (preview.bid && preview.cid) {
+    lastRequested = preview.lastRequested;
+    const {bid, cid} = preview;
+    link = <AdminLink name="collection:records" params={{bid, cid}}>{`${bid}/${cid}`}</AdminLink>;
+  }
+
+  const {lastEditor} = source;
+  return (
+    <ProgressStep {...{label, currentStep, step}}>
+      {lastEditor ? <ReviewInfos {...{active, source, lastRequested, link}}/> : null}
+      {active ? <ReviewButtons onApprove={approveChanges} onDecline={declineChanges}/> : null}
+    </ProgressStep>
+  );
+}
+
+function ReviewInfos({active, source, lastRequested, link} : {active: boolean, source: SourceInfo, lastRequested: number, link: any}) {
+  const {bid, cid, lastEditor, changes={}} = source;
+  const {since, deleted, updated} = changes;
+  return (
+    <ul>
+      <li><strong>Editor: </strong> {lastEditor}</li>
+      <li><strong>Requested: </strong><Date date={lastRequested}/></li>
+      <li><strong>Preview: </strong> {link}</li>
+      {active ?
+       <li>
+         <strong>Changes: </strong>
+         <DiffStats updated={updated} deleted={deleted}/>{" "}
+         <AdminLink name="collection:history"
+                    params={{bid, cid}}
+                    query={{since, resource_name: "record"}}>details...</AdminLink>
+       </li> : null}
+    </ul>
+  );
+}
+
+function DiffStats({updated, deleted} : {updated: number, deleted: number}) {
+  return (
+    <span className="diffstats">
+      {updated > 0 ? <span className="text-green">+{updated}</span> : null}
+      {deleted > 0 ? <span className="text-red">-{deleted}</span> : null}
+    </span>
+  );
+}
+
+function ReviewButtons({onApprove, onDecline} : {onApprove: () => void, onDecline: () => void}) {
+  return (
+    <div className="btn-group">
+      <button className="btn btn-success"
+              onClick={onApprove}>
+        <i className="glyphicon glyphicon-ok"></i> Approve
+      </button>
+      <button className="btn btn-danger"
+              onClick={onDecline}>
+        <i className="glyphicon glyphicon-remove"></i> Decline
+      </button>
+    </div>
+  );
+}
+
+//
+// Signed
+//
+
+function Signed({label,
+                 canEdit,
+                 currentStep,
+                 step,
+                 reSign,
+                 source,
+                 destination} : {label: string,
+                                 canEdit: boolean,
+                                 currentStep: number,
+                                 step: number,
+                                 reSign: () => void,
+                                 source: SourceInfo,
+                                 destination: DestinationInfo}) {
+  const active = (step == currentStep && canEdit);
+  const {lastReviewer} = source;
+  const {lastSigned} = destination;
+  return (
+    <ProgressStep {...{label, currentStep, step}}>
```

Comment:
How about `<ProgressStep label={label} currentStep={currentStep} step={step}>`? Slightly more redundant but much easier to read imho.